.github/workflows/generate_dist_locales.yml

```diff
@@ -56,7 +81,31 @@ jobs:
     - uses: actions/checkout@v4
     - uses: actions/download-artifact@v4
       with:
-        path: dist
+        path: .
+    - name: Cache b3sum
+      id: cache-b3sum
+      uses: actions/cache@v3
+      with:
+        path: /usr/local/bin/b3sum
+        key: ${{ runner.os }}-b3sum-1.5.3
+    - name: Install b3sum
+      if: steps.cache-b3sum.outputs.cache-hit != 'true'
+      run: |
+        curl -L https://github.com/BLAKE3-team/BLAKE3/releases/download/1.5.3/b3sum_linux_x64_bin -o b3sum
+        chmod +x b3sum
+        sudo mv b3sum /usr/local/bin/
+    - name: Verify file integrity
+      run: |
+        rm -f manifest.b3
+        for manifest in $(ls *_manifest.b3 | sort); do
+          echo "${{ secrets.MANIFEST_KEY }}" | b3sum --keyed - "$manifest" > verify.sig
+          cmp -s "$manifest" verify.sig && cat "$manifest" >> manifest.b3
+        done
+
+        if ! b3sum dist --check manifest.b3; then
+          echo "Error: Integrity check failed. Manifest signature or file integrity mismatch."
+          exit 1
+        fi
```

Comment:
I think one potential failure here is if there are locales that don't get updated... though conceptually we should be running this against all locales based off the workflow setup